src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -273,14 +273,15 @@ def iter_index_candidate_infos() -> Iterator[IndexCandidateInfo]:
             )
             icans = list(result.iter_applicable())
 
-            # PEP 592: Yanked releases must be ignored unless only yanked
-            # releases can satisfy the version range. So if this is false,
-            # all yanked icans need to be skipped.
+            # PEP 592: Yanked releases are ignored unless the specifier
+            # explicitely requests a version ('==' or '===') that can be
+            # solely satisfied by a yanked release.
             all_yanked = all(ican.link.is_yanked for ican in icans)
+            direct_specifier = any({'==' in sp.operator for sp in specifier})
```

Comment:
This would allow yanked releases for [compatible release clauses](https://www.python.org/dev/peps/pep-0440/#compatible-release). Something a bit more complicated would be needed.

```python
def is_pinned(specifier: Specifier) -> bool:
    for sp in specifier:
        if sp.operator == "===":
            return True
        if sp.operator != "==":
            continue
        if sp.version.endswith(".*"):
            continue
        return True
    return False
```

Reply:
Done in https://github.com/pypa/pip/pull/10625/commits/8293826bf011d43f54597796738a1a52ef5a6f9d, thank you